src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts

```diff
@@ -0,0 +1,165 @@
+import { FootnoteProps, FootnoteMeta, FootnoteMetaGetters } from './types';
+
+const footnoteFormValidation = ({
+  subjects,
+  indicators,
+  filters,
+  filterGroups,
+  filterItems,
+}: FootnoteProps) => {
+  const errors: { [key: string]: any } = {};
+  const atLeastOneOption =
+    [...subjects, ...indicators, ...filters, ...filterGroups, ...filterItems]
+      .length === 0 &&
+    'At least one Subject, Indicator or Filter must be selected';
+  if (atLeastOneOption) {
+    errors.subjects = atLeastOneOption;
+  }
+  return errors;
+};
+
+export const generateFootnoteMetaMap = (
+  footnoteMeta: FootnoteMeta,
+): FootnoteMetaGetters => {
+  const filtersToSubject: { [key: number]: number } = {};
+  const filterGroupsToFilters: { [key: number]: number } = {};
+  const filterItemsToFilterGroups: { [key: number]: number } = {};
+  const indicatorsToSubject: { [key: number]: number } = {};
+  const indicatorItemsToIndicators: { [key: number]: number } = {};
+
+  Object.keys(footnoteMeta).forEach(subjectId => {
+    Object.keys(footnoteMeta[Number(subjectId)].indicators).forEach(
+      indicatorId => {
+        indicatorsToSubject[Number(indicatorId)] = Number(subjectId);
+
+        Object.keys(
+          footnoteMeta[Number(subjectId)].indicators[Number(indicatorId)]
+            .options,
+        ).forEach(indicatorItemId => {
+          indicatorItemsToIndicators[Number(indicatorItemId)] = Number(
+            indicatorId,
+          );
+        });
+      },
+    );
+
+    Object.keys(footnoteMeta[Number(subjectId)].filters).forEach(filterId => {
+      filtersToSubject[Number(filterId)] = Number(subjectId);
+
+      Object.keys(
+        footnoteMeta[Number(subjectId)].filters[Number(filterId)].options,
+      ).forEach(filterGroupId => {
+        filterGroupsToFilters[Number(filterGroupId)] = Number(filterId);
+
+        Object.keys(
+          footnoteMeta[Number(subjectId)].filters[Number(filterId)].options[
+            Number(filterGroupId)
+          ].options,
+        ).forEach(filterItemId => {
+          filterItemsToFilterGroups[Number(filterItemId)] = Number(
+            filterGroupId,
+          );
+        });
+      });
+    });
+  });
+
+  const getItem = (
+    ids: number[],
+    itemType: 'subject' | 'indicator' | 'filter',
+  ) => {
+    //@ts-ignore
+    if (ids.indexOf(undefined) + 1 || ids.length === 0) {
```

Comment:
Would it be better to use the `Array.includes` function ?

Reply:
Yep, good point :)